package/src/components/Channel/hooks/useCreateChannelContext.ts

```diff
@@ -86,6 +89,7 @@ export const useCreateChannelContext = ({
     [
       channelId,
       disabled,
+      editingDep,
```

Comment:
Why are we adding new dependencies to `ChannelContext` creation ? Why do we pass this here ?

Reply:
(genuine question, need some more context here)

Reply:
We need editing on the top-level channel context, as the message composer should also be able to access the variable. The dep was added so as to respond to editing variable changes in the channel context. If you don't add it, the message composer won't have the value of the editing, so it will have to get us a new composer for editing. Like this is a pattern we have in the SDK for every variable that we use somewhere lower in the hierarchy. Not adding it to the dep will not make the editing work or will the lower component not respond to value change.

We have the editing on the same message input, so we cannot bring it down, regardless, unless we do one more breaking change(with UI).

Reply:
No, I mean why do we need to set it to the `Channel` context at all ? It is obvious that you add it as a dep because it's now part of the context. 

Shouldn't the message composer state allow for setting/unsetting this value elsewhere ? This just means that whenever you enter/exit the editing state everything downstream is going to rerender (because the `ChannelContext` will do this too).

Reply:
The message composer doesn't allow this as for editing it returns a new composer in itself which is populated with the value of that message.

Reply:
Even if that were the case, why do we need it specifically on `Channel` ?

Reply:
Aah, I see. And previously this belonged to the `MessageInputContext` I'm guessing ?

Reply:
Yep, that part is still intact just in case someone wants to access it from there.